ambry-router/src/main/java/com.github.ambry.router/CryptoJobHandler.java

```diff
@@ -69,6 +70,11 @@ public void close() {
           logger.error("Unknown type of job seen : " + task.getClass());
         }
       }
+      try {
+        scheduler.awaitTermination(1000, TimeUnit.MILLISECONDS);
```

Comment:
Could you explain a little why we need wait for 1 sec here?

Reply:
This is trying to fix the earlyCloseReadableChannel test method.
In that particular test, we would close the blobDataReadableChannel before we finish fetching data chunk from server and decrypting them. So there is a chance when the test method is finished, the decryption of some data chunks are still ongoing in the decryption thread. Since in the decrypt job, we still have a ByteBuf not released, we will run into ByteBuf leak error.
So here we wait for some time for all the decrypt jobs to finish and release all the ByteBufs they hold, and we won't have the leak error.